frontend/src/pages/projects/notebook/DeleteNotebookModal.tsx

```diff
@@ -36,10 +43,16 @@ const DeleteNotebookModal: React.FC<DeleteNotebookModalProps> = ({ notebook, onC
           (envFrom) => !envFrom.secretRef?.name.includes(DATA_CONNECTION_PREFIX),
         );
         const configMapNames = nonDataConnectionVariables
-          .filter((envName): envName is ConfigMapRef => !!envName.configMapRef)
+          .filter(
+            (envName): envName is ConfigMapRef =>
+              !!envName.configMapRef && envName.configMapRef.name.startsWith(CONFIGMAP_PREFIX),
```

Comment:
I would not filter by a newly-created configmap or secret name prefix.
Instead, I would suggest by the odh-universal label kex `opendatahub.io/managed-by` which I think you call
https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/const.ts#L5 DATA_CONNECTION_AWS, but it is in fact just indicating that the secret or configmap are managed by opendatahub.

That label would be the ideal one to go for in filtering.

Reply:
> My main argument against prefix-based filtering remains: how can we assume non-dashboard, CICD type external processes do not create secrets that have configmap or secret not as a postfix, but as a prefix, then, they would still be deleted, would they not?

yep and that is the biggest issue here. 

for api call volume we would be fetching for every resource in env from which may be small yes, but could be huge.

Reply:
It could be two-pronged possibly: slow-changing, non-ODH created secrets and configmaps and their info like labels only fetched on first start of dashboard ... and dashboard secrets only fetched by name live, in terms of call volume ...
But then again, that might not be optimal either. Plus, then the logic in dashboard would be more complex, too. I see what you mean by call volume to the API of openshift ...